ui/core/src/utils/time-series-data.ts

```diff
@@ -30,6 +30,38 @@ export function getXValues(timeScale: TimeScale): number[] {
   return xValues;
 }
 
+/**
+ * Given a TimeSeries from a query and a common time scale (see `getCommonTimeScale`),
+ * processes the time series values, filling in any timestamps that are missing
+ * from the time series data with `null` values.
+ */
+export function getTimeSeriesValues(series: TimeSeries, timeScale: TimeScale) {
```

Comment:
Definitely ways we can improve this, but this approach ports how [getYValues](https://github.com/perses/perses/blob/0fcb632ff4e7e4ab3ae60f53dd98886302f9f83f/ui/core/src/utils/time-series-data.ts#L70C17-L84) was implemented before, so now `show_legacy_chart` and the new TimeChart render the same

Reply:
Planning to write better tests for this in a follow up, mocking a similar query to the one where we caught this bug with msw and a visual regression test should do it. Merging for now to get this patch release out quickly

Reply:
Moving this and the getCommonTimeScaleForQueries logic into DataQueriesProvider and merging / processing the data before it gets to TimeSeriesChartPanel is also worth looking into